src/bench/bench.h

```diff
@@ -37,13 +37,13 @@ BENCHMARK(CODE_TO_TIME);
  */
  
 namespace benchmark {
-    // On many systems, the high_resolution_clock offers no better resolution than the steady_clock.
-    // If that's the case, prefer the steady_clock.
+    // In case high_resolution_clock is steady, prefer that, otherwise use steady_clock.
+    // ...but always require at least microsecond precision!
     struct best_clock {
         using hi_res_clock = std::chrono::high_resolution_clock;
         using steady_clock = std::chrono::steady_clock;
-        static constexpr bool steady_is_high_res = std::ratio_less_equal<steady_clock::period, hi_res_clock::period>::value;
-        using type = std::conditional<steady_is_high_res, steady_clock, hi_res_clock>::type;
+        using type = std::conditional<hi_res_clock::is_steady, hi_res_clock, steady_clock>::type;
+        static_assert(std::ratio_less_equal<type::period, std::micro>::value, "steady_clock must have at least microsecond precision");
```

Comment:
Thinking about it, I'm not sure I like the static assert here - do we really prefer the benchmarks to not compile to them compiling with possibly lower precision? Normally a runtime warning would be enough for these kind of things.

Reply:
I was figuring that since its (AFAIK) always the case that the steady_clock has microsecond precision it wasnt a big deal....would use a static_warning if possible, but I guess I could do a runtime print if we prefer that.

Reply:
Don't care whether it's a runtime or compile-time error.
It's just that I already predict complaints coming from someone building on some obscure platform that has lower precision, who probably doesn't even care about the benchmarks but just that the project doesn't build.

Reply:
OK, I moved it to a runtime print.